config/src/config/network_config.rs

```diff
@@ -37,10 +37,13 @@ pub struct NetworkConfig {
     pub discovery_interval_ms: u64,
     pub connectivity_check_interval_ms: u64,
     // Flag to toggle if Noise is used for encryption and authentication.
+    // TODO(philiphayes): remove this flag. no longer useful. all networks must use noise.
     pub enable_noise: bool,
     // If the network uses remote authentication, only trusted peers are allowed to connect.
     // Otherwise, any node can connect. If this flag is set to true, `enable_noise` must
     // also be set to true.
+    // TODO(philiphayes): rename this flag. should reflect `AuthenticationMode` in
+    // `NetworkBuilder`, i.e., `Mutual` vs `ServerOnly` authentication.
     pub enable_remote_authentication: bool,
```

Comment:
we have to be careful here, because ServerOnly is super ambiguous on who is the server and who is the client.

Reply:
Yeah, i'm still not sure what's a good name for not-mutual auth. In my mind, it was always "client-only" auth b/c the dialing peer is the only one meaningfully authenticating the server. But you could also read "client-only" as the client is the only one who is authenticated, so I understand the confusion.

Any suggestions for better names?

Reply:
I think you guys are talking about different things, @davidiw is pointing out that the same code gets called by both clients and servers (but this PR improves things!)

The server-side authentication is correct, but indeed doesn't really describe the situation. Perhaps:

* KnownClient
* UnknownClient

Reply:
in the case of bools, we could have even simpler names

* client_in_trusted_peers
* verify_client

Reply:
EnableRemotePeerAuthentication
if you want a boolean
RemotePeerAuthenticated
RemotePeerUnauthenticated
if you want a an enum
and if we want to have all three in one:
MutualAuthenticated
...